ceph/rados/monitor_workflows.py

```diff
@@ -0,0 +1,260 @@
+        self.client.exec_command(cmd=apply_cmd, sudo=True)
+
+        time.sleep(10)
+        # Checking for the unmanaged setting on mon service
+        cmd = "ceph orch ls"
+        out = self.rados_obj.run_ceph_command(cmd=cmd)
+        for entry in out:
+            if entry["service_name"] == "mon":
+                log.debug(f"Service status : {entry}")
+                status = entry.get("unmanaged", False)
+                if status != unmanaged:
+                    log.error(f"Service not in unmamaned={unmanaged} state. Fail")
+                    return False
+                else:
+                    log.info(f"Service  in unmamaned={unmanaged} state. Pass")
+                    return True
+
+    def check_mon_exists_on_host(self, host) -> bool:
+        """
+        Method to check if the mon daemon exists on the given host
+        Args:
+            host: name of the host where existence of mon needs to be checked
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd = f"ceph orch ps {host}"
+        out = self.rados_obj.run_ceph_command(cmd=cmd)
+        for entry in out:
+            if entry["daemon_type"] == "mon":
+                log.debug("Mon daemon present on the host")
+                return True
+        log.debug("Mon daemon not present on the host")
+        return False
+
+    def remove_mon_service(self, host) -> bool:
+        """
+        Method to remove the mon service from the given host
+
+        Args:
+            host: name of the host from where mon needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mon service exists before removing
+
+        if not self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon not present on the host {host}. Returning Pass")
+            return True
+
+        cmd = f"ceph orch daemon rm mon.{host} --force"
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be removed
+        time.sleep(10)
+
+        if self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon still present on the host {host}. Returning Fail")
+            return False
+
+        # Removing mon label from host if present
+        labels = self.get_host_labels(host=host)
+        if "mon" in labels:
+            cmd = f"ceph orch host label rm {host} mon"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Removed mon label from host : {host}")
+
+        log.debug(f"Mon successfully removed from host : {host}")
+        return True
+
+    def add_mon_service(self, **kwargs) -> bool:
+        """
+        Method to remove the mon service from the given host
+
+        Args:
+            kwargs: the accepted KW args for the method are:
+                host: name of the host from where mon needs to be removed
+                ip: IP of the mon host
```

Comment:
Maybe we could just accept the mon node object and fetch the hostname and ip in the method itself to reduce the number of mandatory arguments.
Very minor though, may not be necessary

Reply:
Done.